axonserver/src/main/java/io/axoniq/axonserver/localstorage/SerializedEvent.java

```diff
@@ -33,34 +33,41 @@
  */
 public class SerializedEvent {
 
-    private final byte[] serializedData;
+    private final DataSupplier dataSupplier;
+
+    public SerializedEvent(DataSupplier dataSupplier) {
+        this.dataSupplier = dataSupplier;
+    }
+
     private volatile Event event;
 
     public SerializedEvent(Event event) {
-        this.serializedData = event.toByteArray();
+        this.dataSupplier = event::toByteArray;
         this.event = event;
     }
 
     public SerializedEvent(byte[] eventFromFile) {
-        this.serializedData = eventFromFile;
+        this.dataSupplier = () -> eventFromFile;
     }
 
     public SerializedEvent(InputStream event) {
-        try {
-            this.serializedData = toByteArray(event);
-        } catch (IOException e) {
-            throw new MessagingPlatformException(ErrorCode.DATAFILE_READ_ERROR, e.getMessage(), e);
-        }
+        this.dataSupplier = () -> {
```

Comment:
This constructor is used while receiving events from the client. If you delay the reading of the stream until it is used, the stream may be closed already by gRPC.

Reply:
also, the input stream can only be read once, so the second time the dataSupplier is called it fails.